pkg/ddl/index.go

```diff
@@ -2549,7 +2555,11 @@ func getNextPartitionInfo(reorg *reorgInfo, t table.PartitionedTable, currPhysic
 		if len(pi.AddingDefinitions) == 0 {
 			// case 1
 			// Simply AddIndex, without any partitions added or dropped!
-			pid, err = findNextPartitionID(currPhysicalTableID, pi.Definitions)
+			if reorg.mergingTmpIdx && currPhysicalTableID == t.Meta().ID {
+				pid = pi.Definitions[0].ID
```

Comment:
I don't understand why the first partition ID is used for global index, could you explain it?

Reply:
If the current Physical id is the table id, the next Physical id should be the first partition id.

Reply:
We cannot distinguish whether this reorg only has the global index, so the partition-related parts must also be scanned.